### PR TITLE
WIP Mal auto updates

### PR DESCRIPTION
**Pull request by cybrox**
Todo:
- Need to actually scenario-test that and sign it off with @toyhammered 
- username/password UI needs some love (maybe a hint about how we store passwords?)
- Maybe add `completed` card

### CODE REVIEW

File: app/templates/components/application/auth-onboarding/export-basicauth.hbs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      {{one-way-text password
+        update=(action (mut password))
+        class="form-control"
+        placeholder="Password"
+        type="password"
+        data-test-selector="password"}}
+    </div>
+  </div>
+  <div class="form-cta">
+    <button type="button" class="button button--primary m-b-1" {{action "changeComponent" "import-select"}}>
```
Comment (trmcnvn) [inline app/templates/components/application/auth-onboarding/export-basicauth.hbs]: `export-select`

**Review by trmcnvn — COMMENTED** [review]
Looks great! ✨ 

I'd like to split the modal out from `application/auth-onboarding` as it isn't apart of the onboarding process.

**Comment by cybrox** [comment]
Granted, would make sense. I think if I do that, I might as well rename all the css classes used to something more general than `import-`.

**Comment by cybrox** [comment]
cc @toyhammered Fixed the type on create (I would still prefer a server-side abstraction for that tbh) and added a remove button. You should be able to scenario-test your implementation like that.

To-Do for me:
- [x] Refactor modals to break them out of onboarding
- [x] Style up remove button and status
- [x] Hide exporter types that are active
- [x] Fix the obvious error Thomas pointed out because I only remembered after this fix commit

**Comment by cybrox** [comment]
This should be pretty much ready to go now.

cc @vevix I added `progressing-modal` as a bit more dynamic abstraction of `auth-onboarding`I think it would make sense to use that as base for the auth modals as well, so we have all the logic for modals with multiple, progressing content pages at one point.

File: app/components/settings/export/export-basicauth.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  actions: {
+    setupExport() {
+      get(this, 'createExport').perform()
+        .then(() => invokeAction(this, 'close'))
+        .catch((err) => {
+          if (get(err, 'errors.firstObject.status') === 422) {
+            get(this, 'notify').error('Authorization failed! Please check your username and password.');
+          } else {
+            get(this, 'notify').error(errorMessages(err));
+          }
```
Comment (trmcnvn) [inline app/components/settings/export/export-basicauth.js]: `get(this, 'raven').logException(err);`

File: app/templates/components/settings/export/export-basicauth.hbs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    <p class="auth-section-helper">Our {{siteName}} exporter will update your anime and manga lists on {{siteName}} every time you update them on Kitsu. Please note that we do <strong>not</strong> delete entries from your {{siteName}} library.</p>
+    <div class="form-group">
+      {{one-way-text username
+        update=(action (mut username))
+        class="form-control"
+        placeholder="Username"
+        autofocus="autofocus"
+        data-test-selector="username"}}
+
+      {{one-way-text password
```
Comment (trmcnvn) [inline app/templates/components/settings/export/export-basicauth.hbs]: `{{one-way-password ...` `type=` can be removed after the change.

File: app/controllers/settings/exports.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import Controller from 'ember-controller';
+
+export default Controller.extend({
+  actions: {
+    removeExport(exporter) {
```
Comment (trmcnvn) [inline app/controllers/settings/exports.js]: Can we move this to the route and use `(route-action "...")` but keep the Controller as a created file due the use of `modelTask/taskValue`.

**Review by trmcnvn — COMMENTED** [review]
Awesome :+1:, I like the progressing-modal approach and will apply it to the auth-modal in a future change.

Just some minor cleanup left! :100:

**Comment by trmcnvn** [comment]
@cybrox is all that's needed testing? Happy to jump in and finish this if you don't have the time.

**Comment by cybrox** [comment]
@vevix Yep, for the code that I added, only proper testing is missing. I won't get around to do much for now, since I'm away on a business trip, so feel free to adjust whatever needed.

Current status of this feature branch is that @toyhammered wanted UI done for actual scenario testing. However, he and @JoshFabian also discussed maby tackling some additions (proper status, history, server-side STI abstraction) before merging. Best ping Toy to get up to date, especially in terms of server side status.

**Comment by toyhammered** [comment]
@vevix here is the pivotal https://www.pivotaltracker.com/story/show/138216731 I should be starting on the backend either today or tomorrow and it won't take me that long. I just need to make sure all the information you need front-end wise is accessible.

**Comment by trmcnvn** [comment]
:+1: I'll follow the pivotal story!

**Comment by toyhammered** [comment]
@JoshFabian sven said that all that is left to do is the displaying of the history (logs). I have made most of my fixes on this open branch: https://github.com/hummingbird-me/hummingbird-server/pull/68

**Pull request closed without merging**